### ISSUE TITLE
Testing Data and Random Seed

### ISSUE THREAD

**Issue by anishmadan23** [ArnoutDevos/maml-cifar-fs#2]
Hi,
I wanted to know that in main.py (Line 293- the comment) if we had meta-batch size as 1, then would doing 600 meta-iterations be equal to choosing 600 tasks?
Also, did you evaluate the test results with different random seed values? I was curious as I was getting significant differences in accuracies in my PyTorch implementation, and I wanted to know if this was observed by you as well.

Thanks

**Comment by ArnoutDevos** [comment]
Hi @anishmadan23 

Thanks for checking out our code and opening this issue.

> I wanted to know that in main.py (Line 293- the comment) if we had meta-batch size as 1, then would doing 600 meta-iterations be equal to choosing 600 tasks?

[Line 293](https://github.com/ArnoutDevos/maml-cifar-fs/blob/56b8dfd367d916d1f5b8ec53ae6c9e747232dc71/main.py#L293) in [main.py](https://github.com/ArnoutDevos/maml-cifar-fs/blob/56b8dfd367d916d1f5b8ec53ae6c9e747232dc71/main.py) makes a comment about how many _different_ tasks (600) will be available in the **meta-validation set** dataloader. See also [line 284](https://github.com/ArnoutDevos/maml-cifar-fs/blob/56b8dfd367d916d1f5b8ec53ae6c9e747232dc71/main.py#L284) which mentions that 200,000 is the number of _different_ tasks available in the **meta-training set** dataloader.

I don't understand your question completely ("choosing 600 tasks"), but I'll give it a try: to have meta-learning you need meta-batch size > 1. This is because you want to meta-learn (i.e. take an outer gradient step) _across_ multiple (>1) tasks. Technically speaking you are right: having meta-batch size = 1 and doing 600 meta-iterations is equal to choosing 600 tasks (1 at every meta-iteration).

> Also, did you evaluate the test results with different random seed values? I was curious as I was getting significant differences in accuracies in my PyTorch implementation, and I wanted to know if this was observed by you as well.

I remember that for different runs (seeds) of this code we obtained similar consistent results (as in our [journal paper](http://rescience.github.io/bibliography/Devos_2019.html)), but didn't report on them (the different runs). Note that taking an average across many test tasks is crucial to reproduce the accuracy results, as performance can vary a lot. Other work ([MAML++](https://arxiv.org/abs/1810.09502)) has shown that MAML is quite sensitive to seed values (at least when re-implemented in PyTorch).

Hope this helps!

**Comment by anishmadan23** [comment]
Thanks for the reply.
I had to apply the 600 meta-iterations to the validation set, hence did not require the outer gradient step. My code was structured in such a way that the difference between choosing tasks and using the dataloader for loading each task created some confusion. So thanks for clarifying that. I will also look at the MAML++ paper for the seed issue.
